src/services/test-helpers/mock/mockApi.ts

```diff
@@ -276,6 +289,187 @@ const referendumInfoOfAt = () =>
+	trieIndex: rococoRegistry.createType('TrieIndex', 60),
+};
+
+const paraLifeCycleObjectOne = {
+	onboarding: true,
+	parachain: true,
+};
+const paraLifeCycleObjectTwo = {
+	parathread: true,
+	parachain: false,
+};
+const paraId1 = typeFactory.storageKey(199);
+const paraId2 = typeFactory.storageKey(200);
+const paraLifecycleOne = rococoRegistry.createType(
+	'ParaLifecycle',
+	paraLifeCycleObjectOne
+);
+const paraLifecycleTwo = rococoRegistry.createType(
+	'ParaLifecycle',
+	paraLifeCycleObjectTwo
+);
+const accountIdOne = rococoRegistry.createType(
+	'AccountId',
+	'1TYrFCWxwHA5bhiXf6uLvPfG6eEvrzzL7uiPK3Yc6yHLUqc'
+);
+const accountIdTwo = rococoRegistry.createType(
+	'AccountId',
+	'13NXiLYYzVEjXxU3eaZNcrjEX9vPyVDNNpURCzK8Bj9BiCWH'
+);
+const balanceOfOne = rococoRegistry.createType('BalanceOf', 1000000);
+const balanceOfTwo = rococoRegistry.createType('BalanceOf', 2000000);
+
+const fundsEntries = () =>
+	Promise.resolve().then(() => {
+		const optionFundInfo = rococoRegistry.createType('Option<FundInfo>', funds);
+
+		const entries = [
+			[paraId1, optionFundInfo],
+			[paraId2, optionFundInfo],
+		];
+
+		return entries;
+	});
+
+const fundsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<FundInfo>', funds);
+	});
+
+const fundsKeys = () =>
+	Promise.resolve().then(() => {
+		return [paraId1, paraId2];
+	});
+
+const parasLifecyclesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, paraLifecycleOne],
+			[paraId2, paraLifecycleTwo],
+		];
+	});
+
+const parasGenesisArgsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('ParaGenesisArgs', { parachain: true });
+	});
+
+const upcomingParasGenesisAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<ParaGenesisArgs>', {
+			parachain: true,
+		});
+	});
+
+const parasLifecyclesAt = () =>
+	Promise.resolve().then(() => {
+		return typeFactory.optionOf(paraLifecycleOne);
+	});
+
+/**
+ * Used for parachain leases
+ */
+const leasesTupleOne = typeFactory.tupleOf(
+	[accountIdOne, balanceOfOne],
+	['AccountId', 'BalanceOf']
+);
+const leasesTupleTwo = typeFactory.tupleOf(
+	[accountIdTwo, balanceOfTwo],
+	['AccountId', 'BalanceOf']
+);
+const parasOptionsOne = typeFactory.optionOf(leasesTupleOne);
+const parasOptionsTwo = typeFactory.optionOf(leasesTupleTwo);
+const vectorLeases = typeFactory.vecOf([parasOptionsOne, parasOptionsTwo]);
+
+export const slotsLeasesAt = (): Promise<Vec<Option<Tuple>>> =>
+	Promise.resolve().then(() => {
+		return vectorLeases;
+	});
+
+const slotsLeasesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, vectorLeases],
+			[paraId2, vectorLeases],
+		];
+	});
+
+/**
+ * Used for parachain Auctions
+ */
+const beingEnd = new BN(1000) as AbstractInt;
+const leasePeriodIndex = new BN(39) as AbstractInt;
+const vectorAuctions = typeFactory.vecOf([beingEnd, leasePeriodIndex]);
```

Comment:
`auctions::AuctionInfo` is a `Option<(LeasePeriodOf<T>, T::BlockNumber)>` so `Option<[beginEnd, leasePeriodIndex]>`. I don't see the `Option` part here. In the service we have it modelled as `Option<Vec<AbstractInt>>` - should wrap this in an `Option`

Reply:
Yea so this is actually wrapped in an Option inside of `auctionInfoAt`, but there were two methods using `vectorAuction's` so I didn't wrap it in an `Option` in the outer scope. But you raised a good point with a previous comment `typeFactory.optionOf(null)` so I fixed it all to be more on point, and now the variable are all inside of `auctionInfoAt` and don't need to be in the outer scope anymore. 

Inside of TypeFactory we now have a `nullOf`, and that allows you too create a Null primitive type and pass it into Option. Good catch and great addition.